lib/python/picongpu/picmi/interaction/interaction.py

```diff
@@ -0,0 +1,154 @@
+
+    @staticmethod
+    def update_constant_list(
+        existing_list: list[pypicongpu.species.constant.Constant],
+        new_list: list[pypicongpu.species.constant.Constant],
+    ) -> None:
+        """check if dicts may be merged without overwriting previously set values"""
+
+        new_constant_list = []
+
+        for constant_new in new_list:
+            exists_already = False
+            for constant in existing_list:
+                if type(constant) == type(constant_new):
+                    # constant_new already exists in existing constants list
+                    exists_already = True
+
+                    if constant != constant_new:
+                        # same type of constant but conflicting values
+                        raise ValueError(f"Constants {constant} and {constant_new} conflict with each other")
+
+            if not exists_already:
+                new_constant_list.append(constant_new)
+            # ignore already existing constants
+
+        # update constant_list
+        existing_list.extend(new_constant_list)
+
+    def get_interaction_constants(
+        self, picmi_species: picmistandard.PICMI_Species
+    ) -> tuple[
+        list[pypicongpu.species.constant.Constant],
+        dict[IonizationModel, pypicongpu.species.constant.ionizationmodel.IonizationModel],
+    ]:
+        """get list of all constants required by interactions for the given species"""
+
+        has_ionization = False
+        constant_list = []
+        ionization_model_conversion = {}
+        for model in self.ground_state_ionization_model_list:
+            if model.ion_species == picmi_species:
+                has_ionization = True
+                model_constants = model.get_constants()
+                Interaction.update_constant_list(constant_list, model_constants)
+                ionization_model_conversion[model] = model.get_as_pypicongpu()
+
+        if has_ionization:
+            # add GroundStateIonization constant for entire species
+            constant_list.append(
+                pypicongpu.species.constant.GroundStateIonization(
+                    ionization_model_list=ionization_model_conversion.values()
+                )
+            )
+
+        # add additional interaction sub groups needing constants here
+        return constant_list, ionization_model_conversion
+
+    def fill_in_ionization_electron_species(
+        self,
+        pypicongpu_by_picmi_species: dict[picmistandard.PICMI_Species, pypicongpu.species.Species],
+        ionization_model_conversion_by_type_and_species: dict[
+            picmistandard.PICMI_Species,
+            None | dict[IonizationModel, pypicongpu.species.constant.ionizationmodel.IonizationModel],
+        ],
+    ) -> None:
+        """
+        add ionization models to pypicongpu species
+
+        in PICMI ioniaztion is defined as a list ionization models owned by the simulation, with each ionization model
```

Comment:
"Owned by A" as in is solely associated with object A, instead of two or more object holding reference to it.

Also the documentation string was somewhat outdated, since the list of ionization models is actually owned by the interaction object which is in turn a member of the simulation object.
Updated the comment to reflect the current state

Reply:
Request was moved to the "will be fixed in future" category and I opened Issue #5139 to keep track of it